app/api/volunteers/needs_api.rb

```diff
@@ -208,6 +211,7 @@ class NeedsApi < Grape::API
                 given_to_id: need.chosen_by.id
               )
 
+              need.update!(status_updated_at: DateTime.now, updated_by: current_user.id)
               need.reviews.create!(review_params)
               need.closed!
```

Comment:
Also here, let's try to use one single update for need and one create for review.